frontend/src/views/SignupView.tsx

```diff
@@ -1,21 +1,39 @@
 import { useNavigate } from "react-router-dom";
-// import UserForm from "../components/UserForm";
+import UserForm from "../components/UserForm";
 
 export default function SignupView() {
     const navigate = useNavigate();
 
-    // const onSubmit = async (
-    //     userid: string,
-    //     password: string,
-    // ): Promise<boolean> => {
-    //     // TODO: if signup is successful, navigate the user to log in
-    //     return false;
-    // };
+    //Signup function to handle the user registration process
+    const onSubmit = async (userId: string, password: string): Promise<boolean> => {
+        try {
+            //API call to create a new account
+            const response = await fetch("api/v1/auth", {
+                method: "POST",
+                headers: {"Content-Type": "application/json"},
+                body: JSON.stringify({userId, password}),
```

Comment:
The auth API uses query params instead of a body bc I'm lazy. So signup should be `POST /api/v1/auth/signup?userid={userid}&password={password}`. Eventually we should actually encrypt and hash the password.

Reply:
I was planning on also having `GET /api/v1/auth/signup/{userid}` check if a userid is available, but if we won't use that I won't implement it.